File: MidiTeslaCoilApp/app/src/main/java/com/example/miditeslacoilapp/ViewModel/BluetoothViewModel.java
package com.example.miditeslacoilapp.ViewModel;

import androidx.lifecycle.ViewModel;

import app.akexorcist.bluetotohspp.library.BluetoothSPP;

public class BluetoothViewModel extends ViewModel {
    private BluetoothSPP bluetoothSPP;

    public void setBluetoothSPP(BluetoothSPP bluetoothSPP) {
        this.bluetoothSPP = bluetoothSPP;
    }

    public BluetoothSPP getBluetoothSPP() {
        return bluetoothSPP;
    }
}

File: MidiTeslaCoilApp/app/src/main/java/com/example/miditeslacoilapp/ViewModel/BluetoothViewModel.kt
package com.example.miditeslacoilapp.ViewModel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import app.akexorcist.bluetotohspp.library.BluetoothSPP

class BluetoothViewModel : ViewModel() {
    private var _bluetoothSPP = MutableLiveData<BluetoothSPP>()
    val bluetoothSPP: LiveData<BluetoothSPP>
        get() = _bluetoothSPP

    fun setBluetooth(bluetoothSPP: BluetoothSPP) = _bluetoothSPP.postValue(bluetoothSPP)

}